pyvrp/cpp/Route.cpp

```diff
@@ -69,7 +69,7 @@ Route::Iterator &Route::Iterator::operator++()
     // we continue to the next until we either exhaust all trips, or we find a
     // non-empty trip.
     ++trip_;
-    while (trip_ + 1 < trips.size() && trips[trip_].empty())
+    while (trip_ < trips.size() && trips[trip_].empty())
```

Comment:
So we don't forget:

`Route::Iterator` provides a flattened view of the clients in a route. A route is composed of one or more trips, so we need to iterate through both the list of trips, and each trip itself. We do this by tracking a `trip_` index, and an `idx_` into `trips[trip_]`.

We should skip empty trips, since indexing those is a big no-no. We do this by checking if the `trips[trip_]` is empty, and then incrementing `trip_` if it is. We should also increment in case we're at the last trip, since that gets us to the iterator returned by `route.end()` and signals end of iteration.

Previously we did not do this - we didn't increment past the last trip. This resulted in the indexing of the last trip, which blows up when that trip is empty. The change here fixes that bug.